index.test-d.ts

```diff
@@ -219,3 +220,16 @@ expectType<ExecaChildProcess<Buffer>>(
 expectType<ExecaChildProcess<Buffer>>(
 	execaNode('unicorns', ['foo'], {nodeOptions: ['--async-stack-traces'], encoding: null}),
 );
+
+expectType<ExecaChildProcess>($`unicorns`);
+expectType<ExecaReturnValue>(await $`unicorns`);
+expectType<ExecaReturnValue>(await $({encoding: 'utf8'})`unicorns`);
+expectType<ExecaReturnValue<Buffer>>(await $({encoding: null})`unicorns`);
+expectType<ExecaReturnValue>(await $({encoding: 'utf8'})`unicorns foo`);
+expectType<ExecaReturnValue<Buffer>>(await $({encoding: null})`unicorns foo`);
+
+expectType<ExecaSyncReturnValue>($.sync`unicorns`);
+expectType<ExecaSyncReturnValue>($({encoding: 'utf8'}).sync`unicorns`);
+expectType<ExecaSyncReturnValue<Buffer>>($({encoding: null}).sync`unicorns`);
+expectType<ExecaSyncReturnValue>($({encoding: 'utf8'}).sync`unicorns foo`);
+expectType<ExecaSyncReturnValue<Buffer>>($({encoding: null}).sync`unicorns foo`);
```

Comment:
We should add some tests with `${...}` to test the type of `TemplateExpression`. With `string`, `number`, `string[]` and `number[]`, and also with another type (like `boolean`), where it should then fail. What do you think?

Reply:
I'm not super familiar with `tsd`. How does this look? https://github.com/sindresorhus/execa/pull/510/commits/656a69b05c9bae4330503f295a8d6d8386ce3fa2

Reply:
That's exactly it. Thank!